### PR TITLE
chore: Detects if L1 constructs are not up-to-date

### PR DESCRIPTION
**Pull request by lantoli**
## Proposed changes

Detects if L1 constructs are not up-to-date

_Jira ticket:_ CLOUDP-251762

Link to any related issue(s): 

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as
  expected)
- [ ] This change requires a documentation update

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run `make fmt` and formatted my code
- [x] I have tested the CDK constructor in a CFN stack. See [TESTING.md](../TESTING.md)
- [x] If changes include removal or addition of 3rd party GitHub actions, I updated our internal document. Reach out to the APIx Integration slack channel to get access to the internal document.

### CODE REVIEW

File: .github/workflows/package.yml [hunk 1]
```diff
@@ -25,23 +20,6 @@ jobs:
         run: yarn install --check-files
       - name: build
         run: npx projen build
-      - name: Find mutations
```
Comment (lantoli) [inline .github/workflows/package.yml:28]: no need to check mutations here as they're already checked in code-health

File: scripts/cdk.sh [hunk 2]
```diff
@@ -54,7 +53,12 @@ cdk-import cfn -l typescript -s "${resourceTemp}" -o "src/l1-resources/${resourc
 # Rename resource file to index.ts file
 dest="src/l1-resources/${resource}/index.ts"
 mv "src/l1-resources/${resource}/${mainFileRoot}.ts" "${dest}"
-python "./scripts/rename_in_file.py" "${dest}"
```
Comment (lantoli) [inline scripts/cdk.sh:57]: python script is deleted now, and its logic is done here with "sed"

File: src/l1-resources/api-key/index.ts [hunk 3]
```diff
@@ -16,11 +16,11 @@ export interface CfnApiKeyProps {
   readonly description: string;
```
Comment (lantoli) [inline src/l1-resources/api-key/index.ts:16]: all changes in src/l1-resources are auto-generated

**Review by oarbusi — APPROVED** [review]
LGTM

**Review by AgustinBettati — APPROVED** [review]
LGTM, this is very nice improvement. One thing to consider is that no all changes in CFN master are always released, in those scenarios we would not want to release CDK with associated changes.

File: .github/workflows/code-health.yml [hunk 4]
```diff
@@ -50,8 +50,38 @@ jobs:
       - uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332
       - name: Run ShellCheck
         uses: bewuethr/shellcheck-action@d01912909579c4b1a335828b8fca197fbb8e0aa4
+  check-l1-updated:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
```
Comment (andreaangiolillo) [inline .github/workflows/code-health.yml:56]: Is `content: read` enough?  you aren't creating a new PR or committing

**Review by andreaangiolillo — APPROVED** [review]
LGTM

File: .github/workflows/code-health.yml (hunk 4 above)
Comment (lantoli) [reply .github/workflows/code-health.yml:56]: thanks @andreaangiolillo for your comment. we're creating an artefact with the patch file when there are mutations, don't we need write for that?

**Comment by lantoli** [comment]
> LGTM, this is very nice improvement. One thing to consider is that no all changes in CFN master are always released, in those scenarios we would not want to release CDK with associated changes.

@AgustinBettati you're right.  Also when a PR is merged in CDK, it's not automatically published in CDK.

So we have to make sure that CDK is released after CFN resources with changes are published

File: .github/workflows/code-health.yml (hunk 4 above)
Comment (andreaangiolillo) [reply .github/workflows/code-health.yml:56]: I don't thinks but I may be wrong, let me know if you found out in the future!

**Pull request merged**